packages/language-support/src/highlighting/syntaxValidation/syntaxValidation.ts

```diff
@@ -132,33 +169,26 @@ export function validateSyntax(
 
   if (wholeFileText.length > 0) {
     const parsingResult = parserWrapper.parse(wholeFileText);
-    const errors = parsingResult.errors;
-
-    if (errors.length === 0) {
-      const semanticAnalysisErrors = doSemanticAnalysis(wholeFileText);
-      semanticAnalysisErrors.forEach((e) => {
-        const start = Position.create(e.line - 1, e.column - 1);
-        const startOffset = e.offset;
-        const end = findEndPosition(parsingResult, start, startOffset);
-
-        errors.push({
-          severity: DiagnosticSeverity.Error,
-          range: {
-            start: start,
-            end: end.relative,
-          },
-          offsets: {
-            start: startOffset,
-            end: end.offset,
-          },
-          message: e.msg,
-        });
-      });
+    diagnostics = parsingResult.diagnostics;
+
+    if (diagnostics.length === 0) {
+      const semanticAnalysisResult = doSemanticAnalysis(wholeFileText);
+      pushToParserDiagnostics(
+        semanticAnalysisResult.errors,
+        DiagnosticSeverity.Error,
+        parsingResult,
+      );
+
+      pushToParserDiagnostics(
+        semanticAnalysisResult.notifications,
+        DiagnosticSeverity.Warning,
+        parsingResult,
+      );
     }
 
     const warnings = warnOnUndeclaredLabels(parsingResult, dbSchema);
-    diagnostics = errors.concat(warnings);
+    diagnostics = diagnostics.concat(warnings);
   }
 
-  return diagnostics;
+  return sortByPosition(diagnostics);
```

Comment:
This is to provide a stable ordering. It's not strictly needed but it would be good for testing purposes to have always the same ordering, especially if we change things in the semantic analysis.